test/fileutils/test_fileutils.rb

```diff
@@ -769,7 +769,7 @@ def test_rm_r_pathname
   def test_rm_r_no_permissions
     check_singleton :rm_rf
 
-    return if /mswin|mingw/ =~ RUBY_PLATFORM
+    return if Process.uid == 0 || /mswin|mingw/ =~ RUBY_PLATFORM
```

Comment:
Probably we'll need to consider capabilities as well.

```suggestion
    return if root_in_posix?
```